src/tests/ftest/fault_domain/fault_domain.py

```diff
@@ -0,0 +1,100 @@
+        :avocado: tags=fault_domain,fault_domain_different_domains
+        """
+        test_passed = True
+        error_messages = list()
+        # Test pools is a list of OrderedDicts
+        # Here is an example of the first element of test_pools:
+        # OrderedDict([('pool_0',
+        #               OrderedDict([('name', 'daos_server'),
+        #                            ('size', '9G'),
+        #                            ('nranks', 1),
+        #                            ('control_method', 'dmg')]))])
+        test_pools = self.params.get("test_pools", "/run/*")
+        fault_paths = self.params.get("fault_path", '/run/*')
+        test_servers = self.hostlist_servers
+
+        for counter, server in enumerate(test_servers):
+            self.add_server_manager()
+            self.configure_manager(
+                "daos_server",
+                self.server_managers[counter],
+                [server],
+                self.hostfile_servers_slots
+            )
+            self.server_managers[counter].set_config_value("fault_path",
+                                                           fault_paths[counter])
+
+        # Servers are started
+        for server in self.server_managers:
+            server.start()
+
+        self.start_agents()
+
+        # Create pools, setting the values obtained from yaml file
+        self.pool = []
+        for index in range(len(test_pools)):
+            pool_yaml_name = "pool_{}".format(index)
+            namespace = test_pools[index]
+            control_method = namespace[pool_yaml_name].get("control_method")
+            nranks = namespace[pool_yaml_name].get("nranks")
+            size = namespace[pool_yaml_name].get("size")
+            self.pool.append(self.get_pool(create=False, connect=False))
+            self.pool[index].control_method.update(name="control_method",
+                                                   value=control_method)
+            self.pool[index].nranks.update(name="nranks", value=nranks)
+            self.pool[index].size.update(name="size", value=size)
+            self.pool[index].create()
+
+        # Once the pools are created we check each pool
+        # and get the ranks used for it.
+        # For each pool:
+        # Using a list, the fault paths of the ranks are added to the same list
+        # The list should not contain any duplicated fault path.
+        for pool in self.pool:
+            pool_fault_path = []
+            for rank in pool.svc_ranks:
+                rank = int(rank)
+                fault_path = fault_paths[rank]
+                pool_fault_path.append(fault_path)
+            if len(pool_fault_path) != len(set(pool_fault_path)):
+                test_passed = False
+                error_message = f"The pool {pool} with ranks {rank} has the" \
+                                f"following fault paths: {pool_fault_path}, " \
+                                f"and must be unique."
```

Comment:
Works for this particular set of tests once I looked at the yaml file, but just to clarify, it is plausible when requesting larger numbers of ranks that some end up under the same domain, particularly when some servers share a fault domain.

Reply:
Could we get a message from dmg/daos server informing the situation? I'm not sure if this should more of a warning or debug information for the administrator.

Reply:
I think the fault domain map would be debug information, maybe something that would be nice to be able to query. The danger level depends a lot on what the container redundancy factors and redundancy levels are. These are set on a per-container basis. If objects are able to be sufficiently replicated across fault domains, there's no real problem.